bundle update rubocop --conservative

skipkayhil opened the pull request:
- fixes a new case of [`Style/RedundantReturn`][1]
- fixes a new case of [`Performance/StringInclude`][2]

[1]: https://github.com/skipkayhil/rails/commit/146b1c2e3389bc70ea0b54abf7843fc1d6c8cd5f
[2]: https://github.com/skipkayhil/rails/commit/3158bbb9f6454dce64dd0b4e2a548351d014c48f

---

When attempting to update Gemfile.lock for Ruby 3.3.0 (#50516), there were a few new warnings raised by RuboCop. This commit splits out just the RuboCop update with fixes for the new warnings introduced.

dhh commented on actionpack/test/controller/routing_test.rb:
```diff
@@ -1694,7 +1694,7 @@ def test_route_with_subdomain_and_constraints_must_receive_params
       get "page/:name" => "pages#show", :constraints => lambda { |request|
         name_param = request.params[:name]
-        return true
+        true
```
Don't like this rule.

skipkayhil replied:
(linking for context) It looks like the rule was enabled in 146b1c2e3389bc70ea0b54abf7843fc1d6c8cd5f after #31004 was merged

dhh replied:
Guess that's why I'd be keen to get Rails under omakase style too.

dhh commented on actionview/lib/action_view/helpers/url_helper.rb:
```diff
@@ -784,7 +784,7 @@ def link_to_remote_options?(options)
 
         def add_method_to_attributes!(html_options, method)
-          if method_not_get_method?(method) && !html_options["rel"]&.match?(/nofollow/)
+          if method_not_get_method?(method) && !html_options["rel"]&.include?("nofollow")
```
This also seems like an odd choice. Maybe fine but it's in the wrong PR.

skipkayhil replied:
(linking for context) `Performance/StringInclude` was enabled in #45865

dhh replied:
Gotcha. Not sure about this one either! But yes, should be in compliance with the rules as they currently are. Just think that needs to happen in a clear PR separate from a dependency bump.

skipkayhil commented:
> Maybe fine but it's in the wrong PR.

I'm not sure I understand the reasoning here. The rules are already being enforced but bumping RuboCop is bringing in bug fixes that cause these two code changes to only be warned on now.

dhh commented:
The dependency bump is broad, bumps all sorts of stuff. The new rubocop rules are specific, and should just get their own explaining PR. Or you can split the dep bump into "just rubocop, plus new rules" and "everything else".

skipkayhil commented:
> The dependency bump is broad, bumps all sorts of stuff. The new rubocop rules are specific, and should just get their own explaining PR. Or you can split the dep bump into "just rubocop, plus new rules" and "everything else".

Got it, the transitive dependencies are being bumped as well but really we should bump the minimum to trigger these violations to fix. I'll make that change 👍

skipkayhil committed:
bundle update rubocop --conservative

Also perform two autocorrects with `bundle exec rubocop -A`:

- fixes a new case of [`Style/RedundantReturn`][1]
- fixes a new case of [`Performance/StringInclude`][2]

[1]: 146b1c2e3389bc70ea0b54abf7843fc1d6c8cd5f
[2]: 3158bbb9f6454dce64dd0b4e2a548351d014c48f

dhh committed:
Merge branch 'main' into pr/50515

* main:
  Add a default yield :head to layouts/application.html.erb (#50527)
  Switch to new enum syntax in example code [ci-skip]
  Utilize new enum syntax
  Apply autocorrection by RuboCop to files generated by `bin/rails generate` (#50506)
  Add brakeman as development dependency
  Fix skipped rubocop test typo (#50519)
  Update the puma startup message
  docs: remove unneeded whitespace from start of code blocks
  Add changelog entry for #50512
  Remove unnecessary require
  Fix railspect not validating alphabetical order
